e2e/npm_translate_lock/package.json

```diff
@@ -1,12 +1,15 @@
 {
     "private": true,
     "dependencies": {
-        "debug": "^4.3.4",
-        "is-odd": "^3.0.1",
-        "semver": "^7.3.8"
+        "debug": "4.3.4",
```

Comment:
its a preference thing IMO; I prefer pinned deps because I know exactly what the inputs to the lock file generator are; the renovator bots for GH will update the deps one way or the other

Reply:
renovator bots will update it either way

Reply:
okay I'll stop picking on you... it just gives developers a false sense of confidence that they think they've pinned the version, and are losing data about whether a given upgrade is actually safe. So it makes me crazy that "most of us are doing it wrong" and I get on my soapbox. It's nice on here!

Reply:
I've been fixing the renovate to stop doing that wrong thing. there's a "preserve semver"  option that we want on. IMO if you have a lockfile, then renovate default should be to leave your dependency ranges alone.

https://docs.renovatebot.com/dependency-pinning/

Reply:
sounds like a longer discussion; I personally I think `^` and `~` are left haphazardly and don't have much meaning anyway since you can't be confident that a bump in version will work or not until renovate makes the PR either way. and in general I think renovate should always bump patch & minor and never major on PRs unless explicitly configured to ignore a certain package in its config. this keeps the logic for what to bump in the renovate config and not in the package.json.

Reply:
This isn't our area of expertise or influence so I think we can just leave it how it is